castBar.lua

```diff
@@ -1,335 +1,397 @@
-local DCB = Dominos:NewModule('CastingBar')
-local CastBar = Dominos:CreateClass('Frame', Dominos.Frame)
-local L = LibStub('AceLocale-3.0'):GetLocale('Dominos-CastingBar')
+--[[
+	CastBar, by Goranaws
+--]]
 
--- Module Code
-function DCB:OnInitialize()
-	_G['CastingBarFrame'].ignoreFramePositionManager = true
-end
+local Addon = _G["Dominos"]
+local Frame = Addon:CreateClass('Frame', Addon.Frame)
+local L = LibStub('AceLocale-3.0'):GetLocale('Dominos-CastingBar')
 
-function DCB:Load()
-	self.frame = CastBar:New()
+--[[ Bar ]]--
+function Frame:New()
+	return Frame.proto.New(self, 'cast')
 end
 
-function DCB:Unload()
-	self.frame:Free()
+local function check(source, target)
+	--you may now add new defaults at will. ~Goranaws
+	if not target then 
+		target = {}
+	end
+	for key, value in pairs(source) do
+		if type(value) == "table" then
+			target[key] =check(value, target[key])
+		else
+			if (type(value) == "boolean") then
+				if target[key] == nil then
+					target[key] = value
+				end
+ 			else
+ 				target[key] = target[key] or value
+ 			end
+		end
+	end
+	return target
 end
 
--- Core Code
-function CastBar:New()
-	local frame = CastBar.proto.New(self, 'cast')
+function Frame:Create(...)
+	local bar = Frame.proto.Create(self, ...)
+	
+	bar.cast = CreateFrame("StatusBar",  bar:GetName().."Bar", bar, "DominosCastingBarFrameTemplate")
```

Comment:
Yes. I'm still working some things out. You can expect another update by
this weekend.
On Apr 17, 2015 7:13 PM, "Jason Greer" <notifications@github.com> wrote:

> In castBar.lua
> <https://github.com/tullamods/Dominos_Cast/pull/2#discussion_r28636896>:
>
> >  end
> >
> > --- Core Code
> > -function CastBar:New()
> > -	local frame = CastBar.proto.New(self, 'cast')
> > +function Frame:Create(...)
> > +	local bar = Frame.proto.Create(self, ...)
> > +	
> > +	bar.cast = CreateFrame("StatusBar",  bar:GetName().."Bar", bar, "DominosCastingBarFrameTemplate")
>
> DominosCastingBarFrameTemplate - Should that be CastingBarFrameTemplate?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tullamods/Dominos_Cast/pull/2/files#r28636896>.
>